### ISSUE TITLE
Provide a way to detect if Python is running from a pup installation

### ISSUE THREAD

**Issue by carlosperate** [mu-editor/pup#237]
I'm looking into a simple way to fix this issue:
- https://github.com/mu-editor/mu/issues/1954

Basically people "pip install mu-editor" in their system Python, and in many Linux distros like Debian and derivatives, that will install any dependencies not already met in the "user site-packages".

Then Mu tries to launch Python with the `-I` (isolated) flag, for things like creating a venv and install pacakges, and it's not able to see packages in the "user site-packages", only the system ones.

Basically, I think we should only use the `-I` flag within Mu if it can detect it's running from a pup installer. So that if a user has installed it via pip or any other package manager, then it run with the same settings that are launching `python -m mu`.

@tmontes would there be a simple way pup can provide a quick check for this? Maybe an environmental variable that is set by pup?
